kernel/kheap.c

```diff
@@ -116,6 +116,25 @@ void *k_heap_alloc(struct k_heap *heap, size_t bytes, k_timeout_t timeout)
 	return ret;
 }
 
+void *k_heap_calloc(struct k_heap *heap, size_t num, size_t size, k_timeout_t timeout)
+{
+	SYS_PORT_TRACING_OBJ_FUNC_ENTER(k_heap, calloc, heap, timeout);
+
+	void *ret = NULL;
+	size_t bounds = 0U;
+
+	if (!size_mul_overflow(num, size, &bounds)) {
+		ret = k_heap_alloc(heap, bounds, timeout);
+	}
+	if (ret != NULL) {
+		(void)memset(ret, 0, bounds);
```

Comment:
This may not work correctly if the allocation alignment is bigger than `(num * size)`. For example, you need 2 bytes but the alignment is 8 bytes (pointer size for 64-bit system), this memset will not clear the last 6 bytes.

Reply:
I see that `k_heap_alloc` is a wrapper for `k_heap_aligned_alloc` with `align` set to  `sizeof(void *)`.
I would use `sizeof(void *)` for memsetting. 
What do you think about it?